src/test/java/org/prebid/server/vertx/http/BasicHttpClientTest.java

```diff
@@ -114,13 +132,111 @@ public void requestShouldFailsIfHttpResponseFails() {
                 .willAnswer(withSelfAndPassObjectToHandler(new RuntimeException("Response exception")));
 
         // when
-        final Future<?> future = httpClient.request(HttpMethod.GET, null, null, null, 0L);
+        final Future<?> future = httpClient.request(HttpMethod.GET, null, null, null, 1L);
 
         // then
         assertThat(future.failed()).isTrue();
         assertThat(future.cause()).hasMessage("Response exception");
     }
 
+    @Test
+    public void requestShouldFailsIfHttpRequestTimedOut(TestContext context) {
+        // given
+        final Vertx vertx = Vertx.vertx();
+        final BasicHttpClient httpClient = new BasicHttpClient(vertx, vertx.createHttpClient());
+        final int serverPort = 7777;
+
+        startServer(context, serverPort, 2000L, 0L);
+
+        // when
+        final Async async = context.async();
+        final Future<?> future = httpClient.get("http://localhost:" + serverPort, 1000L);
+        future.setHandler(ar -> async.complete());
+        async.await();
+
+        // then
+        assertThat(future.failed()).isTrue();
+        assertThat(future.cause())
+                .isInstanceOf(TimeoutException.class)
+                .hasMessageStartingWith("The timeout period of 1000ms has been exceeded");
+    }
+
+    @Test
+    public void requestShouldFailsIfHttpResponseTimedOut(TestContext context) {
+        // given
+        final Vertx vertx = Vertx.vertx();
+        final BasicHttpClient httpClient = new BasicHttpClient(vertx, vertx.createHttpClient());
+        final int serverPort = 8888;
+
+        startServer(context, serverPort, 0L, 2000L);
+
+        // when
+        final Async async = context.async();
+        final Future<?> future = httpClient.get("http://localhost:" + serverPort, 1000L);
+        future.setHandler(ar -> async.complete());
+        async.await();
+
+        // then
+        assertThat(future.failed()).isTrue();
+        assertThat(future.cause())
+                .isInstanceOf(TimeoutException.class)
+                .hasMessage("Timed out while waiting for response");
+    }
+
+    /**
+     * The server returns entire response or body with delay.
+     */
+    private static void startServer(TestContext context, int port, long entireResponseDelay, long bodyResponseDelay) {
+        final Async async = context.async();
```

Comment:
I would say this is misuse of `TestContext`. `CountDownLatch` would be much more suitable here.